### DISCUSSION TITLE
16 digital inputs are not shown from Homematic Wired Module HmIPW-DRI16

### DISCUSSION THREAD

**Discussion by HansHurschle** [SukramJ/aiohomematic#833] [Q&A]
Hello,

i´m new here. 
[config_entry-homematicip_local-b7f25d3e394f96e2715e0d7e6e68621b.json.txt](https://github.com/danielperna84/hahomematic/files/10368601/config_entry-homematicip_local-b7f25d3e394f96e2715e0d7e6e68621b.json.txt)

Over the Christmas holidays I freshly set up a Raspberrymatic with Homematic and Homematic IP hardware components. I have also integrated Homematic IP wired components. The components can be addressed and linked via the Homematic web interface. The Raspberrymatic runs on a Raspberry4 8GB.

On a second, newly installed Raspberry4 8GB now runs Home Assistant. Thanks to your custom integration, all eq-3 Homematic components were found and integrated there as well. However, no inputs are visible for the 16 digital input Homematic Wired Module HmIPW-DRI16. Only the module temperature is visible as an entity. On the same DRAP-bus there is also an 8 channel output module HmIPW-DRS8 where the outputs are visible and controllable.

Many thanks for your support.

**Comment by HansHurschle** [comment]
Sorry - did I mention that I am a newbie?

I see from the linked post in your reply that I need to edit a configuration file to get the entities displayed correctly. I don't have Baxxy13's level of knowledge yet.

I hope maybe to be able to help - however, i need some help myself first, as i am not a programmer, but an electrical engineer and automation engineer. Previously i had a homey and it was issues like this (partially implemented hardware) that had me switching to home assistant now. To get to know the technology in advance, i have this time built a test system with Homematic and Homematic IP components first. So far, however, i see myself only as a user.

Where do I get the information of the hardware, which files do I have to edit and how do i ensure that the findings of the custom database for the device HmIPW-DRI16 then finally benefit the community, so that others do not have to solve the same problem again manually?

Also sorry for deleting the issue template - i don't know how to restore.

**Comment by SukramJ** [comment]
You don't need to edit any configuration files in HA to make this work.

Please ensure that you read the [documantation](https://github.com/danielperna84/custom_homematic#custom_homematic).

I don't own this device so the option names might be different in WebUI
You have 3 options to configure each of your channels in CCUs WebUI:
- KEY (Taster)
- SWITCH (Schalter)
- BINARY (Kontakt)

KEY: key is stateless so no UI element is created and this can be used in a [device trigger](https://www.home-assistant.io/docs/automation/trigger/#device-triggers) within an automation
SWITCH: Creates a switch
BINARY: creates a binary_sensor

Changes in CCUs are applied to HA after a HA restart.

**Comment by SukramJ** [comment]
I migrated this to a discussion as i don't see an issue in this integration.
